app/full_text_search.py

```diff
@@ -144,27 +174,20 @@ def search(repo, query_txt, max_results):
     # which specifies index field name, contains special symbol, ...
     # (e.g., "repo: repository_name", "test: test", "test AND test").
     and_query = romanized_query + ' AND (repo: ' + repo + ')'
+    and_query_with_kanji = romanized_query_with_kanji + ' AND (repo: ' + repo + ')'
 
     person_location_index_results = person_location_index.search(
         appengine_search.Query(
             query_string=and_query, options=options))
 
-    index_results = []
-    regexp = make_or_regexp(query_txt)
-    for document in person_location_index_results:
-        names = ''
-        romanized_jp_names = ''
-        for field in document.fields:
-            if field.name == 'names_romanized_by_romanize_word_by_unidecode':
-                names = field.value
-            if field.name == 'record_id':
-                id = field.value
-            if field.name == 'names_romanized_by_romanize_japanese_name_by_name_dict':
-                romanized_jp_names = field.value
-
-        if regexp.search(names) or regexp.search(romanized_jp_names):
-            index_results.append(id)
+    person_location_index_results_with_kanji = person_location_index.search(
```

Comment:
Add a comment why you need to perform this search in addition to person_location_index_results above.